pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -700,91 +702,84 @@ func (r *Builder) mapNetworks(vm *model.VM, object *cnv.VirtualMachineSpec) (err
-				if err != nil {
-					// Failed to generate network name using template
-					r.Log.Info("Failed to generate network name using template, using default name", "template", networkNameTemplate, "error", err)
 
-					// Fallback to default name and reset error
-					networkName = fmt.Sprintf("net-%v", numNetworks)
-					err = nil
-				}
-			}
+		networkName := fmt.Sprintf("net-%v", numNetworks)
 
-			numNetworks++
-			kNetwork := cnv.Network{
-				Name: networkName,
+		// If a name template is defined, try to use it
+		networkNameTemplate := r.getNetworkNameTemplate(vm)
+		if networkNameTemplate != "" {
+			templateData := api.NetworkNameTemplateData{
+				NetworkName:      mapped.Destination.Name,
+				NetworkNamespace: mapped.Destination.Namespace,
+				NetworkType:      mapped.Destination.Type,
+				NetworkIndex:     numNetworks,
 			}
-			kInterface := cnv.Interface{
-				Name:       networkName,
-				Model:      Virtio,
-				MacAddress: nic.MAC,
+			if generated, err := r.executeTemplate(networkNameTemplate, &templateData); err == nil && generated != "" {
+				networkName = generated
+			} else {
+				r.Log.Info("Failed to generate network name using template, using default", "template", networkNameTemplate, "error", err)
 			}
-			switch mapped.Destination.Type {
-			case Pod:
-				kNetwork.Pod = &cnv.PodNetwork{}
-				kInterface.Masquerade = &cnv.InterfaceMasquerade{}
-			case Multus:
-				kNetwork.Multus = &cnv.MultusNetwork{
-					NetworkName: path.Join(mapped.Destination.Namespace, mapped.Destination.Name),
-				}
-				kInterface.Bridge = &cnv.InterfaceBridge{}
+		}
+
+		numNetworks++
+		kNetwork := cnv.Network{Name: networkName}
+		kInterface := cnv.Interface{
+			Name:       networkName,
+			Model:      Virtio,
+			MacAddress: nic.MAC,
+		}
+
+		switch mapped.Destination.Type {
+		case Pod:
+			kNetwork.Pod = &cnv.PodNetwork{}
+			kInterface.Masquerade = &cnv.InterfaceMasquerade{}
+		case Multus:
+			kNetwork.Multus = &cnv.MultusNetwork{
+				NetworkName: path.Join(mapped.Destination.Namespace, mapped.Destination.Name),
 			}
-			kNetworks = append(kNetworks, kNetwork)
-			kInterfaces = append(kInterfaces, kInterface)
+			kInterface.Bridge = &cnv.InterfaceBridge{}
 		}
+
+		kNetworks = append(kNetworks, kNetwork)
+		kInterfaces = append(kInterfaces, kInterface)
 	}
+
 	object.Template.Spec.Networks = kNetworks
 	object.Template.Spec.Domain.Devices.Interfaces = kInterfaces
 	return
 }
 
+func (r *Builder) findNetworkMapping(nic vsphere.NIC, netMap []api.NetworkPair) (*api.NetworkPair, error) {
+	for i := range netMap {
+		candidate := &netMap[i]
+		network := &model.Network{}
+		if err := r.Source.Inventory.Find(network, candidate.Source); err != nil {
+			continue
+		}
+
+		if (network.Variant == vsphere.NetDvPortGroup || network.Variant == vsphere.OpaqueNetwork) &&
+			nic.Network.ID == network.Key || nic.Network.ID == network.ID {
+			return candidate, nil
+		}
+	}
+	return nil, ErrNoNetworkMapping
```

Comment:
Don't think that we need to return the error, we already check if the map is nil and then continue. Plust there are no other errors from this function